step-maven-plugin/src/main/java/step/plugins/maven/ExecuteAutomationPackageMojo.java

```diff
@@ -200,16 +190,20 @@ protected ControllerCredentials getControllerCredentials() {
         return new ControllerCredentials(getUrl(), authToken == null || authToken.isEmpty() ? null : authToken);
     }
 
-    protected MultitenancyClient createMultitenancyClient() {
-        return new RemoteMultitenancyClientImpl(getControllerCredentials());
-    }
-
     protected RemoteAutomationPackageClientImpl createRemoteAutomationPackageClient() {
-        return new RemoteAutomationPackageClientImpl(getControllerCredentials());
+        RemoteAutomationPackageClientImpl client = new RemoteAutomationPackageClientImpl(getControllerCredentials());
+        addProjectHeaderToRemoteClient(client);
```

Comment:
To avoid the need to call the method addProjectHeaderToRemoteClient after each instantiation, I was wondering if it would make sense to add the field project or a generic map for the context attributes to the controller credentials

Reply:
@jeromecomte 
Not sure that the ControllerCredentials is the best place to keep the map with additional headers. But I've changed the simple map in the `AbstractRemoteClient` to the wrapping `AdditionalHeaders` to add some default headers (tenant header) easier.